### PR TITLE
feat!: plot svg layers

### PR DESCRIPTION
**Pull request by wadjih-bencheikh18**
closes : https://github.com/zakodium/react-plot/issues/353

### CODE REVIEW

File: src/utils/splitChildren.tsx [hunk 1]
```diff
@@ -12,7 +12,8 @@ import { RangeSeries } from '../components/Series/RangeSeries';
 import { ScatterSeries } from '../components/Series/ScatterSeries';
 
 interface PlotChildren {
-  seriesAndAnnotations: ReactElement[];
+  series: ReactElement[];
+  annotations: ReactElement | null;
```
Comment (targos) [inline src/utils/splitChildren.tsx:16]: annotations should be an array too. It's allowed to have multiple groups of annotations.

File: src/components/Plot.tsx [hunk 2]
```diff
@@ -211,72 +212,99 @@ export function Plot(props: PlotProps) {
...
-            />
-
-            {/* Series viewport */}
-            <g transform={`translate(${leftOffset}, ${topOffset})`}>
+              style={{
+                ...finalSvgStyle,
+                position: 'absolute',
+                top: '0',
+                left: '0',
+                zIndex: '10',
```
Comment (targos) [inline src/components/Plot.tsx:226]: I think if you swap the elements' zIndex becomes unnecessary?

File: src/utils/splitChildren.tsx (hunk 1 above)
Comment (wadjih-bencheikh18) [reply src/utils/splitChildren.tsx:16]: okey

File: src/components/Plot.tsx (hunk 2 above)
Comment (wadjih-bencheikh18) [reply src/components/Plot.tsx:226]: going to try

Comment (wadjih-bencheikh18) [reply src/components/Plot.tsx]: fixed

**Review by stropitek — COMMENTED** [review]
The 2d spectrum story still feels slow though, but it's because of scripting, not rendering

image

**Comment by targos** [comment]
> The 2d spectrum story still feels slow though, but it's because of scripting, not rendering
> 
> image

It's slow because of https://github.com/zakodium/react-plot/issues/357

File: src/components/Plot.tsx [hunk 3]
```diff
@@ -211,72 +212,98 @@ export function Plot(props: PlotProps) {
...
-            </g>
+                <legendOffsetContext.Provider value={legendOffset}>
+                  <g ref={legendBbox.ref}>{legend}</g>
+                </legendOffsetContext.Provider>
+              </g>
 
-            <g ref={headingBbox.ref}>{heading}</g>
-          </svg>
+              <g ref={headingBbox.ref}>{heading}</g>
+            </svg>{' '}
```
[the remark this answers is not in the document]
Comment (wadjih-bencheikh18) [reply src/components/Plot.tsx]: fixed

**Pull request merged**